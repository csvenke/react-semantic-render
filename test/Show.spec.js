import React from "react";
import renderer from "react-test-renderer";

import { Show as Show1 } from "../dist/index.cjs";
import { Show as Show2 } from "../dist/index.esm";
import Show3 from "../dist/Show";
import Show4 from "../src/Show";

const data = {
  input: <div>render me</div>,
  output: { type: "div", props: {}, children: ["render me"] }
};

const tests = [
  {
    description:
      "should return result if when is true and children is JSX.element",
    props: {
      when: true,
      children: data.input,
      render: null
    },
    result: data.output
  },
  {
    description:
      "should return result if when is false and children is JSX.element",
    props: {
      when: false,
      children: data.input,
      render: null
    },
    result: null
  },
  {
    description:
      "should return result if when is true and render returns JSX.element",
    props: {
      when: true,
      children: null,
      render: () => data.input
    },
    result: data.output
  },
  {
    description:
      "should return result if when is false and render returns JSX.element",
    props: {
      when: false,
      children: null,
      render: () => data.input
    },
    result: null
  },
  {
    description: "should not evaluate render if when is false",
    props: {
      when: false,
      children: null,
      render: () => doesNotExist.a.b.c
    },
    result: null
  },
  {
    description:
      "should return content from children if when is true and both children and render is used",
    props: {
      when: true,
      children: data.input,
      render: () => <div>do not render me!</div>
    },
    result: data.output
  },
  {
    description:
      "should return null if when is true and both children and render is used",
    props: {
      when: false,
      children: data.input,
      render: () => <div>do not render me!</div>
    },
    result: null
  },
  {
    description:
      "should return null if when is true and both children and render is not used",
    props: {
      when: true,
      children: null,
      render: null
    },
    result: null
  },
  {
    description: "should return null if when is true and render returns null",
    props: {
      when: true,
      children: null,
      render: () => null
    },
    result: null
  },
  {
    description: "should return result if when is true and children is string",
    props: {
      when: true,
      children: "Hello World",
      render: null
    },
    result: "Hello World"
  },
  {
    description:
      "should return result if when is true and render returns string",
    props: {
      when: true,
      children: null,
      render: () => "Hello World"
    },
    result: "Hello World"
  }
];

const testCases = [
  {
    description: "dist/index.cjs",
    component: Show1,
    tests
  },
  {
    description: "dist/index.esm",
    component: Show2,
    tests
  },
  {
    description: "dist/Show",
    component: Show3,
    tests
  },
  {
    description: "src/Show",
    component: Show4,
    tests
  }
];

describe.each(testCases)("Show", ({ description, component, tests }) => {
  const Show = component;
  describe.each(tests)(`${description}`, ({ description, props, result }) => {
    it(`${description}`, () => {
      const element = <Show {...props} />;
      const component = renderer.create(element);
      expect(component.toJSON()).toEqual(result);
    });
  });
});
