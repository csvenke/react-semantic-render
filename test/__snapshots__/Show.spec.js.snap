// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with children and render should return null if when equals false 1`] = `null`;

exports[`with children and render should return null if when equals false 2`] = `null`;

exports[`with children and render should return null if when equals false 3`] = `null`;

exports[`with children and render should return null if when equals false 4`] = `null`;

exports[`with children and render should return primary content from children and ignore render if when equals true 1`] = `
<div>
  children
</div>
`;

exports[`with children and render should return primary content from children and ignore render if when equals true 2`] = `
<div>
  children
</div>
`;

exports[`with children and render should return primary content from children and ignore render if when equals true 3`] = `
<div>
  children
</div>
`;

exports[`with children and render should return primary content from children and ignore render if when equals true 4`] = `
<div>
  children
</div>
`;

exports[`with children should return null if when equals false 1`] = `null`;

exports[`with children should return null if when equals false 2`] = `null`;

exports[`with children should return null if when equals false 3`] = `null`;

exports[`with children should return null if when equals false 4`] = `null`;

exports[`with children should return primary content from children if when equals true 1`] = `
<div>
  render me
</div>
`;

exports[`with children should return primary content from children if when equals true 2`] = `
<div>
  render me
</div>
`;

exports[`with children should return primary content from children if when equals true 3`] = `
<div>
  render me
</div>
`;

exports[`with children should return primary content from children if when equals true 4`] = `
<div>
  render me
</div>
`;

exports[`with render should not evaluate render if when is undefined 1`] = `null`;

exports[`with render should not evaluate render if when is undefined 2`] = `null`;

exports[`with render should not evaluate render if when is undefined 3`] = `null`;

exports[`with render should not evaluate render if when is undefined 4`] = `null`;

exports[`with render should return null if when equals false 1`] = `null`;

exports[`with render should return null if when equals false 2`] = `null`;

exports[`with render should return null if when equals false 3`] = `null`;

exports[`with render should return null if when equals false 4`] = `null`;

exports[`with render should return primary content from render if when equals true 1`] = `
<div>
  render me
</div>
`;

exports[`with render should return primary content from render if when equals true 2`] = `
<div>
  render me
</div>
`;

exports[`with render should return primary content from render if when equals true 3`] = `
<div>
  render me
</div>
`;

exports[`with render should return primary content from render if when equals true 4`] = `
<div>
  render me
</div>
`;

exports[`without children and render should return null 1`] = `null`;

exports[`without children and render should return null 2`] = `null`;

exports[`without children and render should return null 3`] = `null`;

exports[`without children and render should return null 4`] = `null`;
